src/quarot/gptq.py

```diff
@@ -23,23 +23,36 @@ def quantize_weight_gptq(
     clip_weights: bool = True,
 ) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor | None]:
     """
-    TODO.
+    Quantize a weight tensor to INT<bits> using the GPTQ scheme.
+
+    Args:
+        W: The weight tensor to quantize.
+        H: The Hessian of the activations, must have dtype==torch.float32.
+        bits: The number of bits to quantize to.
+        symmetric: Whether to use symmetric quantization.
+        max_blocksize: The maximum block size for the GPTQ algorithm.
+        percdamp: The damping factor for the Hessian.
+        groupsize: The number of weights to group together for quantization.
+        clip_weights: Whether to clip the weights (by searching for good clipping ratios) to the range of the quantization.
     """
     num_features, num_columns = W.shape
     orig_dev = W.device
     orig_dtype = W.dtype
-    W = W.to('cuda').float()
-    H = H.cuda()
+    W = W.to('cuda', dtype=torch.float32)
+    H = H.to(W.device)
 
     if groupsize is None:
         scale, offset = calculate_scales(W, bits, symmetric=symmetric, clip_weights=clip_weights, vectorized=False)
         scale = scale.float()
-        offset = offset.float() if offset is not None else None
+        if not symmetric:
+            offset = offset.float()
+    else:
+        raise NotImplementedError("Groupwise quantization not yet tested for GPTQ")
 
     # find dead weights and set them to zero, and their Hess value to 1
-    dead = torch.diag(H) == 0
-    H[dead, dead] = 1
-    W[:, dead] = 0
+    dead = torch.diag(H) == 0.0
+    H[dead, dead] = 1.0
+    W[:, dead] = 0.0
```

Comment:
perhaps we could log the number of dead weights? it should surely be zero for any modern model!